public/app/features/explore/CorrelationEditorModeBar.tsx

```diff
@@ -0,0 +1,137 @@
+        setShowSavePrompt(true);
+      } else if (correlationDetails?.dirty === false) {
+        // otherwise, if we are exiting in a not dirty state, reset everything
+        setShowSavePrompt(false);
+        dispatch(changeCorrelationDetails({ label: undefined, description: undefined, canSave: false }));
+        panes.forEach((pane) => {
+          dispatch(removeCorrelationData(pane[0]));
+          dispatch(runQueries({ exploreId: pane[0] }));
+        });
+      }
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [correlationsEditorMode, correlationDetails?.dirty]);
+
+  return (
+    <>
+      {/* Handle navigating outside of Explore */}
+      <Prompt
+        message={(location, action) => {
+          if (location.pathname !== '/explore' && correlationsEditorMode && (correlationDetails?.dirty || false)) {
+            return 'You have unsaved correlation data. Continue?'
+          } else {
+            return true;
+          }
+        }}
+      />
+
+      {showSavePrompt && (
+        <CorrelationUnsavedChangesModal
+          onDiscard={() => {
+            // if we are discarding the in progress correlation, reset everything
+            dispatch(changeCorrelationDetails({ dirty: false }));
+          }}
+          onCancel={() => {
+            // if we are cancelling the exit, set the editor mode back to true and hide the prompt
+            dispatch(changeCorrelationsEditorMode({ correlationsEditorMode: true }));
+            setShowSavePrompt(false);
+          }}
+          onSave={() => {
+            dispatch(saveCurrentCorrelation(correlationDetails?.label, correlationDetails?.description));
+            dispatch(changeCorrelationDetails({ dirty: false }));
+          }}
+        />
+      )}
+      {toShow && (
+        <div className={styles.correlationEditorTop}>
+          <HorizontalGroup spacing="md" justify="flex-end">
+            <Tooltip content="Correlations editor in Explore is an experimental feature.">
+              <Icon name="info-circle" size="xl" />
+            </Tooltip>
+            <Button
+              variant="secondary"
+              disabled={!correlationDetails?.canSave}
+              fill="outline"
+              onClick={() => {
+                dispatch(saveCurrentCorrelation(correlationDetails?.label, correlationDetails?.description));
+              }}
+            >
+              Save
+            </Button>
+            <Button
+              variant="secondary"
+              fill="outline"
+              icon="times"
+              onClick={() => {
+                dispatch(changeCorrelationsEditorMode({ correlationsEditorMode: false }));
+              }}
+              aria-label="exit correlations editor mode"
+            >
+              Exit Correlation Editor
+            </Button>
+          </HorizontalGroup>
+        </div>
+      )}
+    </>
+  );
+};
+
+const getStyles = (theme: GrafanaTheme2) => {
```

Comment:
The bar looks off after scrolling to the bottom:

<img width="1536" alt="Screenshot 2023-09-04 at 14 04 23" src="https://github.com/grafana/grafana/assets/745532/0f3f0f9e-f0b1-48de-b561-05110615b1bc">

Reply:
Fixed!